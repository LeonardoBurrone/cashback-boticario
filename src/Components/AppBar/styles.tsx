import { AppBar } from '@material-ui/core';
import { Theme } from '@material-ui/core/styles';
import styled from 'styled-components';

type Props = {
  theme: Theme;
};

const drawerWidth = 200;

const CBAppBar = styled(AppBar)`
  && {
    z-index: ${(props: Props) => props.theme.zIndex.drawer + 1};
    transition: ${(props: Props) =>
      props.theme.transitions.create(['width', 'margin'], {
        duration: props.theme.transitions.duration.leavingScreen,
        easing: props.theme.transitions.easing.sharp
      })};
    &.appBarShift {
      margin-left: ${drawerWidth}px;
      width: calc(100% - ${drawerWidth}px);
      transition: ${(props: Props) =>
        props.theme.transitions.create(['width', 'margin'], {
          duration: props.theme.transitions.duration.enteringScreen,
          easing: props.theme.transitions.easing.sharp
        })};
    }
  }
`;

const CBButtonDiv = styled('div')`
  && {
    display: flex;
    flex-grow: 1;
    justify-content: flex-end;
  }
`;

const LogoImg = styled('img')`
  && {
    width: 100px;
  }
`;

export { CBAppBar, CBButtonDiv, LogoImg };
