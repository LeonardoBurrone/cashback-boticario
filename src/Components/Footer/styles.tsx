import { Link } from 'react-router-dom';
import styled from 'styled-components';

const CBContainer = styled('div')`
  && {
    display: flex;
    flex-direction: column;
    margin: 30px -20px -20px -20px;
  }
`;

const CBGrid = styled('div')`
  && {
    display: flex;
    justify-content: flex-end;
    padding: 24px 24px;
  }
`;

const CBLink = styled(Link)`
  && {
    text-decoration: none;
  }
`;

export { CBContainer, CBGrid, CBLink };
