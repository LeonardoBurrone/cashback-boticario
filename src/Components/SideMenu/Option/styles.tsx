import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import styled from 'styled-components';

const OptionWrapper = styled(ListItem)`
  && {
    max-width: 200px;
    height: 70px;
    padding: 2px 10px 2px 0px;
    position: relative;
  }

  &.is-active {
    max-width: 210px;
    border-top-right-radius: 6px;
    border-bottom-right-radius: 6px;
  }
`;

const OptionIcon = styled(ListItemIcon)`
  && {
    opacity: 0.7;
    min-width: 20px;
  }
`;

const OptionText = styled(ListItemText)`
  && {
    min-width: 130px;
    white-space: normal;
    & .text-primary {
      font-size: 14px;
    }
    & .text-secondary {
      opacity: 0.7;
      font-size: 12px;
    }
  }
`;

export { OptionIcon, OptionText, OptionWrapper };
