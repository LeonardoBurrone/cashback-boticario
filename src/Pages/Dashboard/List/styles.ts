import styled from 'styled-components';

const Information = styled('div')`
  && {
    align-items: baseline;
    display: flex;
    justify-content: space-between;
  }
`;

export { Information };
