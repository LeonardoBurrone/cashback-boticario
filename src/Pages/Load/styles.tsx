import { Dialog } from '@material-ui/core';
import styled from 'styled-components';

const CBDialog = styled(Dialog)`
  & .MuiDialog-paperScrollPaper {
    align-items: center;
    padding: 30px;
  }
`;

export { CBDialog };
