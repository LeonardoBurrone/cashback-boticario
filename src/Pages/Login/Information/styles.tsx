import { Paper, TextField } from '@material-ui/core';
import styled from 'styled-components';

const CBContainer = styled(Paper)`
  && {
    padding: 20px;
    &.center {
      display: flex;
      flex-direction: column;
      justify-content: center;
      align-items: center;
    }
  }
`;

const CBTextField = styled(TextField)`
  && {
    min-width: 100%;
  }
`;

const Title = styled.p`
  align-items: flex-start;
  display: flex;
  flex-flow: column;
`;

export { CBContainer, CBTextField, Title };
