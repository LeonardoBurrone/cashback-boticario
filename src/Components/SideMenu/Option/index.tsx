/* eslint-disable react/no-children-prop */
import { Icon } from '@material-ui/core';
import { useTheme } from '@material-ui/core/styles';
import React from 'react';
import { Link, Route } from 'react-router-dom';

import { OptionIcon, OptionWrapper, OptionText, TextDiv } from './styles';

import { MenuOption } from '../types';

import { checkActivePath } from '../../../Services/BrowserHistory';

export type Props = {
  isSideMenuOpened: boolean;
  option: MenuOption;
};

export const Option: React.FunctionComponent<Props> = (props: Props) => {
  const theme = useTheme();

  return (
    <Route
      path={props.option.path}
      children={({ match }) => {
        const isActive = checkActivePath(match);

        return (
          <Link color="secondary" to={props.option.path} style={{ textDecoration: 'none' }}>
            <OptionWrapper button className={isActive ? 'is-active' : ''} theme={theme}>
              <OptionIcon className={isActive ? 'is-active' : ''} theme={theme}>
                <Icon>{props.option.icon}</Icon>
              </OptionIcon>
              {props.isSideMenuOpened && (
                <TextDiv>
                  <OptionText className={isActive ? 'is-active' : ''} theme={theme} variant="subtitle1">
                    {props.option.title}
                  </OptionText>
                  <OptionText className={isActive ? 'is-active' : ''} theme={theme} variant="caption">
                    {props.option.description}
                  </OptionText>
                </TextDiv>
              )}
            </OptionWrapper>
          </Link>
        );
      }}
    />
  );
};

export default Option;
