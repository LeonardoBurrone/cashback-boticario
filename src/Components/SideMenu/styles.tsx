import { Drawer } from '@material-ui/core';
import { Theme } from '@material-ui/core/styles';
import styled from 'styled-components';

type Props = {
  theme: Theme;
};

const drawerWidth = 200;

const CBDrawer = styled(Drawer)`
  && {
    width: ${drawerWidth}px;
    flex-shrink: 0;
    whitespace: 'nowrap';
    &.drawerOpen,
    &.drawerOpen .MuiDrawer-paper {
      width: ${drawerWidth}px;
      transition: ${(props: Props) =>
        props.theme.transitions.create(['width'], {
          duration: props.theme.transitions.duration.enteringScreen,
          easing: props.theme.transitions.easing.sharp
        })};
    }
    &.drawerClose,
    &.drawerClose .MuiDrawer-paper {
      transition: ${(props: Props) =>
        props.theme.transitions.create(['width'], {
          duration: props.theme.transitions.duration.leavingScreen,
          easing: props.theme.transitions.easing.sharp
        })};
      width: ${(props: Props) => props.theme.spacing(7.5)}px;
      ${(props: Props) => props.theme.breakpoints.up('sm')} {
        width: ${(props: Props) => props.theme.spacing(7.5)}px;
      }
      overflow-x: hidden;
    }
  }
`;

const OpenButtonDiv = styled('div')`
  && {
    display: flex;
    alignitems: center;
    justify-content: flex-end;
    padding: ${(props: Props) => props.theme.spacing(0, 1)};
  }
`;

export { CBDrawer, OpenButtonDiv };
