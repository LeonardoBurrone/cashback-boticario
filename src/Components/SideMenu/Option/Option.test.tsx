import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import { Route } from 'react-router-dom';

import Option, { Props } from './';
import { OptionIcon, OptionWrapper, OptionText } from './styles';

jest.mock('../../../Services/BrowserHistory', () => ({
  checkActivePath: jest.fn().mockReturnValueOnce(false).mockReturnValue(true)
}));

describe('Test component Option', () => {
  const props: Props = {
    isSideMenuOpened: true,
    option: {
      description: 'Veja a listagem de suas compras cadastradas',
      icon: 'assessment',
      path: '/dashboard',
      title: 'Dashboard'
    }
  };

  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<Option {...props} />);
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it('Check if component rendered correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Validate layout when option is not active', () => {
    const routeProps: any = wrapper.find(Route).props();
    const LinkComponent = () => routeProps.children({ match: '' });
    const linkWrapper = shallow(<LinkComponent />);

    expect(linkWrapper.find(OptionWrapper).props().className).toBe('');
    expect(linkWrapper.find(OptionIcon).props().className).toBe('');
    expect(linkWrapper.find(OptionText).at(0).props().className).toBe('');
    expect(linkWrapper.find(OptionText).at(1).props().className).toBe('');
  });

  it('Validate layout when option is active', () => {
    const routeProps: any = wrapper.find(Route).props();
    const LinkComponent = () => routeProps.children({ match: '' });
    const linkWrapper = shallow(<LinkComponent />);

    expect(linkWrapper.find(OptionWrapper).props().className).toBe('is-active');
    expect(linkWrapper.find(OptionIcon).props().className).toBe('is-active');
    expect(linkWrapper.find(OptionText).at(0).props().className).toBe('is-active');
    expect(linkWrapper.find(OptionText).at(1).props().className).toBe('is-active');
  });
});
