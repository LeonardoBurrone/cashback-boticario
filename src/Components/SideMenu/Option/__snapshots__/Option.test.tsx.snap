// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component Option Check if component rendered correctly 1`] = `
<Route
  path="/dashboard"
>
  <Component />
</Route>
`;
